config/src/config/test_data/validator_full_node.yaml

```diff
@@ -26,14 +26,17 @@ full_node_networks:
               token:
                   from_disk: "/full/path/to/token"
       network_id: "public"
-      max_frame_size: 8388608 # 8 MiB
     - listen_address: "/ip4/0.0.0.0/tcp/7180"
       network_id:
           private: "vfn"
       seed_addrs:
           "c227da54069989f283712e4016704660":
               - "/ip4/127.0.0.1/tcp/58259/ln-noise-ik/c998dcd54c3daf48e0ad516d94b7be0b0b7a27caa00541f2b2c14b13500df10b/ln-handshake/0"
+    - discovery_method: "onchain"
+      listen_address: "/ip4/0.0.0.0/tcp/8180"
+      network_id: "public"
 
 upstream:
     networks:
         - private: vfn
+        - public
```

Comment:
cc: @mgorven for config changes

Reply:
What does this config change do? Why is it needed?

Reply:
tl;dr: this adds a public network for a node to failover to for state sync/mempool if the first private vfn network goes down

all networks listed here are 'upstream' networks, which means a node will request state sync from / broadcast txns to this peers in this network. The networks are ordered in preference, so adding the public network allows a node to be able to fallback to it if the first (=highest preferred) network is unavailable

Reply:
So this means that a VFN will try to sync state from whatever random public full nodes are connected to it? Is it expected that running a public fullnode potentially means providing outgoing state sync bandwidth to VFNs?

Reply:
public network here is going to be the validators advertised on-chain (which is their VFN), so the VFN will in reality fallback to another VFN

Reply:
Oh, there are two `public` networks defined in this config file, that doesn't seem right. The network with other VFNs should be named differently.

Reply:
this is due to our **beautiful** network model, I don't think we can change the name since one of the public actually connects to another public (in a different node) (they handshake the network id) lol